src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -93,6 +94,12 @@ void CancelWork()
                                 builders[runtimeIndex] = ArrayBuilder<ActiveStatementDebugInfo>.GetInstance(pendingStatements);
                                 builders[runtimeIndex].Count = pendingStatements;
 
+                                if (instructionMap.Count == 0)
+                                {
+                                    completion.TrySetResult(builders.ToFlattenedImmutableArrayAndFree());
```

Comment:
This also needs to do `if (Interlocked.Decrement(ref pendingRuntimes) == 0)`

Reply:
We escape after if (Interlocked.Decrement(ref pendingRuntimes) == 0)

in lines 80-83 if errorCode != 0 
and 
in lines 142-145 if there are no pendingStatements.

Why should we decrement pendingRuntimes here?

---
In reply to: [199300592](https://github.com/dotnet/roslyn/pull/28070#discussion_r199300592) [](ancestors = 199300592)

Reply:
Because this return only returns from lambda that processes a single runtime. We should not complete the task if there are more runtimes to process.

---
In reply to: [200480800](https://github.com/dotnet/roslyn/pull/28070#discussion_r200480800) [](ancestors = 200480800,199300592)